In [7]:
from io import BytesIO
import requests
import pandas as pd
import numpy as np

In [8]:
req = requests.get('https://docs.google.com/spreadsheets/d/e/2PACX-1vR-ti6Su94955DZ4Tky8EbwifpgZf_dTjpBdiVH0Ukhsq94jZdqoHuUytZsFZKfwpXEUCKRFteJRc9P/pub?gid=889004448&single=true&output=csv')
data = req.content

df = pd.read_csv(BytesIO(data))

In [3]:
df.head()

date                time event platform   ad_id ad_cost_type  ad_cost
0  2019-04-01  2019-04-01 0:00:48  view      web  121288          CPM    187.4
1  2019-04-01  2019-04-01 0:04:41  view      ios  121288          CPM    187.4
2  2019-04-01  2019-04-01 0:07:50  view  android  121288          CPM    187.4
3  2019-04-01  2019-04-01 0:07:50  view  android  121288          CPM    187.4
4  2019-04-01  2019-04-01 0:08:46  view      ios  121288          CPM    187.4

В данных вы найдете информацию о событиях, которые произошли с объявлением 121288 за два дня. Рассчитайте следующие метрики в разрезе каждого дня:

- количество показов
- количество кликов
- CTR 
- сумма потраченных денег \
То есть для каждой метрики у вас должно получиться два числа - за 2019-04-01 и 2019-04-02

Рассчитать сумму потраченных денег можно по следующей формуле - разделите значение из колонки ad_cost на 1000 и умножьте на количество показов объявления

In [9]:
df.query("event == 'view'")

date                 time event platform   ad_id ad_cost_type  \
0    2019-04-01   2019-04-01 0:00:48  view      web  121288          CPM   
1    2019-04-01   2019-04-01 0:04:41  view      ios  121288          CPM   
2    2019-04-01   2019-04-01 0:07:50  view  android  121288          CPM   
3    2019-04-01   2019-04-01 0:07:50  view  android  121288          CPM   
4    2019-04-01   2019-04-01 0:08:46  view      ios  121288          CPM   
..          ...                  ...   ...      ...     ...          ...   
579  2019-04-02  2019-04-01 23:47:39  view  android  121288          CPM   
580  2019-04-02  2019-04-01 23:54:58  view      web  121288          CPM   
581  2019-04-02  2019-04-01 23:55:05  view  android  121288          CPM   
582  2019-04-02  2019-04-01 23:55:22  view  android  121288          CPM   
583  2019-04-01   2019-04-01 0:08:46  view      ios  121288          CPM   

     ad_cost  
0      187.4  
1      187.4  
2      187.4  
3      187.4  
4      187.4  
..       ...  
579    187.4  
580    187.4  
581    187.4  
582    187.4  
583    187.4  

[584 rows x 7 columns]

In [19]:
len(df.query("event == 'click' and date == '2019-04-02'"))/len(df.query("event == 'view' and date == '2019-04-02'")) \
-len(df.query("event == 'click' and date == '2019-04-01'"))/len(df.query("event == 'view' and date == '2019-04-01'"))

0.029892911109651137

In [31]:
# df['ad_cost_'] = df['ad_cost']/1000
df.ad_cost_.unique()[0]*len(df.query("event == 'view' and date == '2019-04-02'")) \
# /df.ad_cost_.unique()[0]*len(df.query("event == 'view' and date == '2019-04-01'"))

17.4282

In [32]:
17.4282/92.0134

0.18940936863543786

In [29]:
df.ad_cost_.unique()[0]*len(df.query("event == 'view' and date == '2019-04-01'"))

92.0134

In [18]:
len(df.query("event == 'click' and date == '2019-04-02'"))/len(df.query("event == 'view' and date == '2019-04-02'")) 

0.06451612903225806

In [11]:
len(df.query("event == 'click'"))

23